lib/matplotlib/tests/test_category.py

```diff
@@ -187,6 +187,37 @@ def test_plot_1d_missing(self):
         self.axis_test(ax.yaxis, self.dmticks, self.dmlabels, self.dmunit_data)
 
     @cleanup
+    @pytest.mark.usefixtures("data")
+    @pytest.mark.parametrize("bins",
+                             [['a', 'b', 'c', 'a'],
+                              [b'a', b'b', b'c', b'a'],
+                              np.array([b'a', b'b', b'c', b'a'])])
+    def test_plot_bytes(self, bins):
+        counts = np.array([4, 6, 5, 1])
+
+        fig, ax = plt.subplots()
+        ax.bar(bins, counts)
+        fig.canvas.draw()
+
+        self.axis_test(ax.xaxis, self.dticks, self.dlabels, self.dunit_data)
+
+    @cleanup
+    @pytest.mark.parametrize("bins",
+                             [['1', '11', '3', '1'],
+                              np.array(['1', '11', '3', '1']),
+                              [b'1', b'11', b'3', b'1'],
+                              np.array([b'1', b'11', b'3', b'1'])])
+    def test_plot_numlike(self, bins):
+        counts = np.array([4, 6, 5, 1])
```

Comment:
Maybe I should have named it `bars` instead of `bins`... The bars just overlap each other. Actually, it's probably not necessary; I just used the categories from the original issue.